hedera-node/src/main/java/com/hedera/services/state/logic/StandardProcessLogic.java

```diff
@@ -86,42 +86,64 @@ public StandardProcessLogic(
 	public void incorporateConsensusTxn(SwirldTransaction platformTxn, Instant consensusTime, long submittingMember) {
 		try {
 			final var accessor = expandHandleSpan.accessorFor(platformTxn);
-			Instant effectiveConsensusTime = consensusTime;
-			if (accessor.canTriggerTxn()) {
-				final var offset = dynamicProperties.triggerTxnWindBackNanos();
-				effectiveConsensusTime = consensusTime.minusNanos(offset);
-			}
 
-			if (!invariantChecks.holdFor(accessor, effectiveConsensusTime, submittingMember)) {
+			if (!invariantChecks.holdFor(accessor, consensusTime, submittingMember)) {
 				return;
 			}
 
-			sigImpactHistorian.setChangeTime(effectiveConsensusTime);
-			expiries.purge(effectiveConsensusTime.getEpochSecond());
+			consensusTimeTracker.reset(consensusTime);
+
+			sigImpactHistorian.setChangeTime(consensusTime);
+			expiries.purge(consensusTime.getEpochSecond());
 			sigImpactHistorian.purge();
 			recordStreaming.resetBlockNo();
 
-			doProcess(submittingMember, consensusTime, effectiveConsensusTime, accessor);
+			doProcess(submittingMember, consensusTimeTracker.isFirstUsed()
```

Comment:
It still feels confusing that client code needs to use this
```
consensusTimeTracker.isFirstUsed()
    ? consensusTimeTracker.nextTransactionTime(true)
    : consensusTimeTracker.firstTransactionTime()
```
idiom...can we just have `consensusTimeTracker.nextAvailableForTriggeringTransaction()`?

Reply:
I was thinking the caller should explicitly be aware of where it allows being the first transaction and where it would only be after the first one.